### PR TITLE
Fully support JuliaArtifacts.toml filename everywhere.

### PR DESCRIPTION
**Pull request by fredrikekre**
---

Currently, e.g. adding this fork of `JuliaBinaryWrappers/Zlib_jll.jl` with `Artifacts.toml` changed to `JuliaArtifacts.toml`
```
pkg> add https://github.com/fredrikekre/Zlib_jll.jl
```
doesn't download artifacts at install time.

### CODE REVIEW

**Comment by fredrikekre** [comment]
bors try

**Comment by fredrikekre** [comment]
bors try

File: src/Artifacts.jl [hunk 1]
```diff
@@ -370,12 +373,12 @@ returns the `Platform` object that this entry specifies.  Returns `nothing` on e
...
 function unpack_platform(entry::Dict, name::String, artifacts_toml::String)
     if !haskey(entry, "os")
-        @error("Invalid Artifacts.toml at '$(artifacts_toml)': platform-specific artifact entry '$name' missing 'os' key")
+        @error("Invalid (Julia)Artifacts.toml at '$(artifacts_toml)': platform-specific artifact entry '$name' missing 'os' key")
         return nothing
     end
 
     if !haskey(entry, "arch")
-        @error("Invalid Artifacts.toml at '$(artifacts_toml)': platform-specific artifact entrty '$name' missing 'arch' key")
+        @error("Invalid (Julia)Artifacts.toml at '$(artifacts_toml)': platform-specific artifact entrty '$name' missing 'arch' key")
```
[the remark this answers is not in the document]
Comment (fredrikekre) [reply src/Artifacts.jl]: Okay, how about
```
Invalid artifacts file at ...
```
?

[the remark this answers is not in the document]
Comment (staticfloat) [reply src/Artifacts.jl]: Even better.

**Pull request merged**